### PR TITLE
refactoring and added parallelism

### PR DESCRIPTION
**Pull request by neumann-dev**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [x] I have tested the changes.
- [x] This is not a duplicate.
- [x] I have read the README file and the description.
- [x] I have added a description which explains this pull request.
- [x] All new and existing tests passed.

## Description

Added parallelism with streams and refactored the code to an more objective-oriented way

### CODE REVIEW

File: src/ansc/ANSChecker.java [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+		 */
+		DragLayout dl_dragPanel = new DragLayout();
+		dl_dragPanel.setUsePreferredSize(false);
+		this.dragPanel = new JPanel(dl_dragPanel);
+		this.frmAnimeNewSeason.getContentPane().add(new JScrollPane(dragPanel));
+
+		/*
+		 * Now all elements in the JFrame are being set and added to the DragLayout.
+		 */
+		this.tfAnimeHeadline = initTextField(dragPanel,
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Format code here so it is in one line instead of multiple lines.

File: src/ansc/ANSChecker.java [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+				COLOR_GREEN,
+				SwingConstants.CENTER,
+				10, 10, 502, 44,
+				FONT_TEXTFIELD,
+				false,
+				"Anime:",
+				10);
+
+		this.tfAnimeSeasonHeadline = initTextField(dragPanel,
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Format code here so it is in one line instead of multiple lines.

File: src/ansc/ANSChecker.java [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+				COLOR_GREEN,
+				SwingConstants.CENTER,
+				522, 10, 160, 44,
+				FONT_TEXTFIELD,
+				false,
+				"Newest season?",
+				10);
+
+		this.tfAnimeRunHeadline = initTextField(dragPanel,
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Format code here so it is in one line instead of multiple lines.

File: src/ansc/ANSChecker.java [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+				COLOR_GREEN,
+				SwingConstants.CENTER,
+				692, 10, 400, 44,
+				FONT_TEXTFIELD,
+				false,
+				"Anime runtime:",
+				10);
+
+		this.tfControls = initTextField(dragPanel,
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Format code here so it is in one line instead of multiple lines.

File: src/ansc/ANSChecker.java [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+				COLOR_GREEN,
+				SwingConstants.CENTER,
+				1102, 10, 274, 44,
+				FONT_TEXTFIELD,
+				false,
+				"Controls:",
+				10);
+
+		this.tfSeason = initTextField(dragPanel,
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Format code here so it is in one line instead of multiple lines.

File: src/ansc/ANSChecker.java [hunk 6]
```diff
@@ -0,0 +1,369 @@
...
+		 * Search button triggers the check() function and adds the content to the
+		 * JTextFields if it is successful.
+		 */
+		JButton btnSearch = new JButton("Search for new seasons");
+		btnSearch.setFont(new Font("Tahoma", Font.PLAIN, 15));
+		btnSearch.setBounds(1102, 164, 274, 44);
+		btnSearch.addActionListener(e -> checkButtonPressed());
+		this.dragPanel.add(btnSearch);
+
+		this.tfAnimeCount = initTextField(dragPanel,
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Format code here so it is in one line instead of multiple lines.

File: src/ansc/ANSChecker.java [hunk 7]
```diff
@@ -0,0 +1,369 @@
...
+				COLOR_CYAN,
+				SwingConstants.CENTER,
+				1102,214, 274, 44,
+				FONT_BUTTON,
+				false,
+				"You have " + this.animeList.size() + " anime titles listed!",
+				10);
+
+		this.tfDuration = initTextField(dragPanel,
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Format code here so it is in one line instead of multiple lines.

File: src/ansc/ANSChecker.java [hunk 8]
```diff
@@ -0,0 +1,369 @@
...
+	}
+
+	private void initAnimeListGuiElements() {
+		/*
+		 * JTextFields are being created and added to DragLayout for all anime titles
+		 * which are in the .txt file.
+		 */
+		// counter is the same, but you only need one instead of one for each
+		int i = 0;
+		for (Anime anime : this.animeList) {
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Add whitespace before for.

File: src/ansc/ANSChecker.java [hunk 9]
```diff
@@ -0,0 +1,369 @@
...
+			initTextField(this.dragPanel,
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Format code here so it is in one line instead of multiple lines.

File: src/ansc/ANSChecker.java [hunk 10]
```diff
@@ -0,0 +1,369 @@
...
+							COLOR_LIGHTGREY,
+							null,
+							10, 64 + i, 502, 44,
+							FONT_ANIME_TF,
+							false,
+							anime.getName(),
+							10);
+
+			// Maps are <Key, Value>-Pairs, so u can get the TF of an anime with his name
+			this.tfAnimeSeasonMap.put(anime.getName(),
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Format code here so it is in one line instead of multiple lines.

File: src/ansc/ANSChecker.java [hunk 11]
```diff
@@ -0,0 +1,369 @@
...
+					initTextField(this.dragPanel,
+							COLOR_LIGHTGREY,
+							null,
+							522, 64 + i, 160, 44,
+							FONT_ANIME_TF,
+							false,
+							null,
+							10));
+
+			this.tfAnimeRunMap.put(anime.getName(),
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Format code here so it is in one line instead of multiple lines.

File: src/ansc/ANSChecker.java [hunk 12]
```diff
@@ -0,0 +1,369 @@
...
+							FONT_ANIME_TF,
+							false,
+							null,
+							10));
+
+			i += 50;
+		}
+	}
+
+	private JTextField initTextField(JPanel dragPanel,
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Format code here so it is in one line instead of multiple lines.

File: src/ansc/ANSChecker.java [hunk 13]
```diff
@@ -0,0 +1,369 @@
...
+									 Integer boundY,
+									 Integer boundWidth,
+									 Integer boundHeight,
+									 Font font,
+									 Boolean editable,
+									 String text,
+									 Integer numberOfColumns) {
+		JTextField textField = new JTextField();
+		textField.setBackground(backgroundColor);
+		if (horizontalAlignment != null)
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Add whitespace before if and add whitespace after statement.

File: src/ansc/ANSChecker.java [hunk 14]
```diff
@@ -0,0 +1,369 @@
...
+			textField.setHorizontalAlignment(horizontalAlignment);
+		textField.setBounds(boundX, boundY, boundWidth, boundHeight);
+		textField.setFont(font);
+		textField.setEditable(editable);
+		if (text != null)
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Add whitespace before if and add whitespace after statement.

File: src/ansc/ANSChecker.java [hunk 15]
```diff
@@ -0,0 +1,369 @@
...
+	 */
+	private void check() {
+		this.startTime = System.nanoTime();
+
+		this.animeList = this.animeList.parallelStream().peek(anime -> {
+			Document website;
+			try {
+				website = Jsoup.connect(anime.getUrl()).get();
+			} catch (IOException e) {
+				JOptionPane.showMessageDialog(null, "Error: " +
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Format code here so it is in one line instead of multiple lines.

File: src/ansc/ANSChecker.java [hunk 16]
```diff
@@ -0,0 +1,369 @@
...
+						anime.getName() + " not available!");
+				return;
+			}
+
+			// Format: "%d seasons"
+			Element seasonElement = website.getElementById("browse-episodes-season");
+
+			String seasonData;
+			// When seasonElement is null there is only one season
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Add whitespace above comment.

File: src/ansc/ANSChecker.java [hunk 17]
```diff
@@ -0,0 +1,369 @@
...
+				seasonData = seasonElement.attr("aria-label");
+			} else {
+				seasonData = "1 season";
+			}
+
+			Integer numberOfSeasons = Integer.valueOf((seasonData.split(" "))[0]);
+			Integer expectedNumberOfSeasons = Integer.valueOf(this.tfSeason.getText());
+
+			String seasonText;
+			if (numberOfSeasons.equals(expectedNumberOfSeasons)) {
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Add whitespace before if and add whitespace after else statement.

File: src/ansc/ANSChecker.java [hunk 18]
```diff
@@ -0,0 +1,369 @@
...
+				seasonText = "Season " + numberOfSeasons + " newest!";
+			} else {
+				seasonText = numberOfSeasons.equals(1) ? "1 season!" : numberOfSeasons + " seasons!";
+			}
+			anime.setSeason(seasonText);
+
+
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Remove one whitespace.

File: src/ansc/ANSChecker.java [hunk 19]
```diff
@@ -0,0 +1,369 @@
...
+			String yearData;
+			if (website.select("#browse-episodes-year.ipc-simple-select__input").text().contains("See all")) {
```
Comment (mexikoedi) [inline src/ansc/ANSChecker.java]: Add whitespace before if.

File: src/model/AnimeListReader.java [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+
+        while ((textLine = br.readLine()) != null) {
+            String animeName;
+            String animeURL;
+
+            if (textLine.lastIndexOf("anime:") >= 0) {
+                animeName = textLine.substring(textLine.lastIndexOf(":") + 1);
+
+                textLine = br.readLine();
+                if (textLine.lastIndexOf("source:") >= 0) {
```
Comment (mexikoedi) [inline src/model/AnimeListReader.java]: Add whitespace before if.

**Review by mexikoedi — CHANGES_REQUESTED** [review]
I noticed some small code style stuff.
See my comments.

**Comment by neumann-dev** [comment]
the code style stuff is done

**Review by mexikoedi — APPROVED** [review]
Looks good to me.
I will do one or two changes here and there but in overall good.

**Pull request merged**